lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -910,6 +913,39 @@ export class UserAgentApplication {
     return adalFrame;
   }
 
+  private pollWindowForHash(contentWindow: Window): Promise<string> {
+      return new Promise((resolve, reject) => {
+          const intervalTimeMs = 50;
+          const maxTicks = this.config.system.loadFrameTimeout / intervalTimeMs;
+          let ticks = 0;
+
+          const intervalId = setInterval(() => {
+              try {
+                  // Will throw if cross origin
+                  const href = contentWindow.location.href;
+
+                  // Don't process blank pages
+                  if (href === "" || href === "about:blank") {
+                      return;
+                  }
+
+                  // Only start clock when we are on same domain
+                  ticks++;
+
+                  if (contentWindow.closed) {
+                      clearInterval(intervalId);
+                 } else if (this.urlContainsHash(href)) {
+                      clearInterval(intervalId);
+                      resolve(contentWindow.location.hash);
```

Comment:
How does this work in a sandboxed iframe? Does this not come under scripts?

Reply:
This will be running in the parent window, which has access to the location object if the iframe is on the same domain.